pkg/test/framework/scope.go

```diff
@@ -32,6 +33,8 @@ type scope struct {
 
 	resources []resource.Resource
 
+	closers []io.Closer
```

Comment:
Nit: You can model this as a resource, as you did before, just not exposed (and required) as a component.

Reply:
I this I prefer this approach, even though it requires a second array. It's a bit odd to go through the assignment of a resource ID for a closer lambda.